src/library/implementation_interface.hpp

```diff
@@ -163,8 +163,16 @@ struct ParallelImplementation {
 
     static void close(){};
     static std::tuple<hid_t, hid_t> prepare_write(const std::string& report_name, hid_t plist_id) {
-        // Enable MPI access
         MPI_Info info = MPI_INFO_NULL;
+
+        // Disable MPI-IO optimizations for better IME support
+        if (report_name.find("ime://") == 0 || report_name.find("/ime/") == 0) {
```

Comment:
(comment : I was wondering if there is any programming way to say this directory is IME specific but that's not a worth for now)

Reply:
Thanks Pramod, this is actually a very good point. I was wondering the same after our discussion yesterday, because if we mount a folder ".../scratch/projXX/ime" or similar, the trick wouldn't work.

Let me check the documentation, just in case there is a way to deduct that a given path is on IME.

Reply:
@pramodk There is now an `IMEUtil` class that provides an utility to check whether a file is on IME or not, and if so, convert the path to the proper absolute path necessary for the IME Native API. I have also moved there the MPI-IO hints setting, instead of having this functionality somewhere else.